### PR TITLE
Nklofutar invitations and new task assignees

### PR DESCRIPTION
**Pull request by nevenklofutar**
New Tasks, TaskAssignees, Invitations

### CODE REVIEW

File: server/src/server.ts [hunk 1]
```diff
@@ -18,7 +17,6 @@ app.use(urlencoded({ extended: true }));
 
 // Guards
 app.use('/api', authorizationGuard);
-app.use('/api/tasks/:taskid', taskAssigneeGuard);
```
Comment (mihokrusic) [inline server/src/server.ts:21]: Why did you remove this?

File: server/src/controllers/api/invitation.controller.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { Router } from 'express';
+import { create, update, remove, findByInviterId, findByInviteeId } from '../../services/invitation.service';
+const router = Router();
+
+router.get('/byInviterId', async (req, res) => {
```
Comment (mihokrusic) [inline server/src/controllers/api/invitation.controller.ts:5]: /inviter

File: server/src/controllers/api/invitation.controller.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            task: invitation.task,
+        }));
+
+        res.status(200).json(invitationsView);
+    } else {
+        res.status(400).json({ message: result.message });
+    }
+});
+
+router.get('/byInviteeId', async (req, res) => {
```
Comment (mihokrusic) [inline server/src/controllers/api/invitation.controller.ts:22]: /invitee

File: server/src/controllers/api/invitation.controller.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        };
+
+        res.status(200).json(invitationsView);
+    } else {
+        res.status(400).json({ message: result.message });
+    }
+});
+
+// TODO: we dont need this ? noone will call this ?
+router.delete('/byTaskId', async (req, res) => {
```
Comment (mihokrusic) [inline server/src/controllers/api/invitation.controller.ts:82]: I agree, we dont need this, remove it

File: server/src/db/invitation.model.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import mongoose, { Schema } from 'mongoose';
+import { UserDocument } from './user.model';
+import { TaskDocument } from './task.model';
+
+export enum InvitationStatus {
+    Pending = 1,
```
Comment (mihokrusic) [inline server/src/db/invitation.model.ts:6]: Remove explicit numbers, by default Typescript enums start with 0

File: server/src/db/task-assignee.model.ts [hunk 6]
```diff
@@ -1,18 +1,23 @@
 import mongoose, { Schema } from 'mongoose';
+import { UserDocument } from './user.model';
+
+export enum Role {
+    Creator = 1,
```
Comment (mihokrusic) [inline server/src/db/task-assignee.model.ts:5]: Remove explicit numbers, by default Typescript enums start with 0

File: server/src/services/task.service.ts [hunk 7]
```diff
@@ -9,19 +12,16 @@ export const find = async (userId, sort) => {
...
-        .in(taskIds);
+    // TODO: find by creator only or by any of the assignees ?
+    const query = TaskModel.find({ 'assignees.assignee': userId });
 
     if (sort.field) {
         query.sort(`${sort.dir === 'desc' ? '-' : ''}${sort.field}`);
     }
 
-    return query.exec();
+    const result = await query.exec();
```
Comment (mihokrusic) [inline server/src/services/task.service.ts:22]: Why two lines? Why not ```return await query.exec()```

File: server/src/server.ts (hunk 1 above)
Comment (nevenklofutar) [reply server/src/server.ts:21]: we don't need it anymore ? i'm using user in all queries ?

File: server/src/services/task.service.ts (hunk 7 above)
Comment (nevenklofutar) [reply server/src/services/task.service.ts:22]: my bad, i was prob testing stuff so it stayed like this after i removed testing

File: server/src/db/invitation.model.ts (hunk 5 above)
Comment (nevenklofutar) [reply server/src/db/invitation.model.ts:6]: I like explicit numbers more tbh, by fine.

File: server/src/db/task-assignee.model.ts (hunk 6 above)
Comment (nevenklofutar) [reply server/src/db/task-assignee.model.ts:5]: I like explicit numbers more tbh, by fine.

File: server/src/server.ts (hunk 1 above)
Comment (nevenklofutar) [reply server/src/server.ts:21]: we need a guard for taskitems

File: server/src/controllers/api/invitation.controller.ts [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+        res.status(200).json(invitationsView);
+    } else {
+        res.status(400).json({ message: result.message });
+    }
+});
+
+router.post('/', async (req, res) => {
+    const inviter = (req as any).user.id;
+    const invitee = req.body.invitee;
```
Comment (mihokrusic) [inline server/src/controllers/api/invitation.controller.ts:41]: You can do ```const { invitee, task } = req.body```

File: server/src/controllers/api/invitation.controller.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    const task = req.body.task;
+
+    // TODO: this needs a guard or something - create invitation where current user is in assignees
```
Comment (mihokrusic) [inline server/src/controllers/api/invitation.controller.ts:44]: Add it :)

File: server/src/controllers/api/invitation.controller.ts [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+});
+
+router.put('/:invitationId', async (req, res) => {
+    const { invitationId } = req.params;
+    const inviterId = (req as any).user;
+    const status = req.body.status;
+
+    const result = await update(invitationId, inviterId, status);
+    if (result.valid) {
+        const invitationsView = {
```
Comment (mihokrusic) [inline server/src/controllers/api/invitation.controller.ts:68]: Can we extract this map into a separate function here? It's repeated several times

File: server/src/controllers/api/invitation.controller.ts (hunk 9 above)
Comment (nevenklofutar) [reply server/src/controllers/api/invitation.controller.ts:43]: Solved it by checking if inviter is in assigness of the task

File: server/src/server.ts (hunk 1 above)
Comment (nevenklofutar) [reply server/src/server.ts:21]: created queries that include assigneess so no need for guard
